script/proto-docs.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+set -eux -o pipefail
+
+mkdir -p docs/grpc
+
+DOCROOT="${DOCROOT:$PWD}"
+target=docs/grpc
+templates=docs/grpc-templates
+filemap="$templates/filemap.xml"
+
+cp "$templates/index-header.html" "$target/index.html"
+
+
+for pkg in \
+    'api/events' \
+    'api/types' \
+    'api/types/task' \
+    'api/services/images/v1/' \
+    'api/services/tasks/v1/' \
+    'api/services/content/v1/' \
+    'api/services/snapshot/v1/' \
+    'api/services/namespaces/v1/' \
+    'api/services/diff/v1/' \
+    'api/services/version/v1/' \
+    'api/services/events/v1/' \
+    'api/services/leases/v1/' \
+    'api/services/introspection/v1/' \
+    'api/services/containers/v1/' \
+; do
+    protoc \
+        --doc_out="root=$DOCROOT,filemap=$filemap:$target" \
```

Comment:
Just curious, which plugin are we using for this? https://github.com/pseudomuto/protoc-gen-doc doesn't seem to have any of this options

Reply:
https://github.com/dnephin/prototools/ (link is in the PR description), forked from https://github.com/sourcegraph/prototools (more details also in the PR description).

I tried pseudomuto/protoc-gen-doc as well. It generated docs, but they were cluttered with a bunch of external extensions. It's possible I had the config wrong, or we could fix it with custom templates, but `prototools` seemed to be closer to what I expected. Also 
https://github.com/pseudomuto/protoc-gen-doc/issues/319 is an issue with that plugin as well, so either way we will need to patch it.

Reply:
I’d rather we forked or write out that is properly behaved. The issue with the source graph tool is that it has these xml preamble files that don’t make a whole lot of sense to adopt in this project.

Reply:
The xml is just a config for the plugin. It should be really easy to replace it with something else.